src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunctionSet.cs

```diff
@@ -98,6 +103,35 @@ internal TexlFunctionSet(IEnumerable<TexlFunction> functions)
             }
         }
 
+        // This will compose multiple function sets into a single one. 
+        // The assumption is that all incoming function sets are unchanged for the lifetime of the returned set.
+        // Hence it doesn't matter where this shares references or copies the values. 
+        internal static TexlFunctionSet Compose(IEnumerable<TexlFunctionSet> functionSets)
```

Comment:
This is critical. 
Each symbol table has a function set, but most are empty. 
It's also key that the caller of this method knows the symbols won't change on us - this is true because this is only called during a binding pass (which operates on a fixed set of symbols - and has safety checks to throw if the symbols are changed during binding).